ql/src/java/org/apache/hadoop/hive/ql/history/HiveScribeImpl.java

```diff
@@ -0,0 +1,332 @@
+package org.apache.hadoop.hive.ql.history;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.hadoop.hive.ql.MapRedStats;
+import org.apache.hadoop.hive.ql.QueryPlan;
+import org.apache.hadoop.hive.ql.QueryStats;
+import org.apache.hadoop.hive.ql.exec.Task;
+import org.apache.hadoop.hive.ql.session.SessionState;
+import org.apache.hadoop.mapred.Counters;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HiveScribeImpl implements HiveHistory {
+  private static final Logger LOG = LoggerFactory.getLogger("hive.ql.exec.HiveScribeImpl");
+
+  String histFileName; // History file name
+
+  private static final Random randGen = new Random();
+
+  private SessionState.LogHelper console;
+
+  private Map<String, String> idToTableMap = null;
+
+  // Job Hash Map
+  private final HashMap<String, HiveHistory.QueryInfo> queryInfoMap = new HashMap<String, HiveHistory.QueryInfo>();
+
+  // Task Hash Map
+  private final HashMap<String, HiveHistory.TaskInfo> taskInfoMap = new HashMap<String, HiveHistory.TaskInfo>();
+
+  private static final String DELIMITER = " ";
+
+  private static final String ROW_COUNT_PATTERN = "RECORDS_OUT_(\\d+)(_)*(\\S+)*";
+
+  private static final Pattern rowCountPattern = Pattern.compile(ROW_COUNT_PATTERN);
+
+  // Query Hash Map
+  public static HashMap<String, QueryStats> queryStatsMap = new HashMap<String, QueryStats>();
+
+  /**
+   * Construct HiveScribeImpl object and open history log file.
+   *
+   * @param ss
+   */
+  public HiveScribeImpl(SessionState ss) {
+    LOG.info("Instantiated an instance for HiveScribeImpl. Replaced hive history file at local dir\n");
+    HashMap<String, String> hm = new HashMap<String, String>();
+    hm.put(HiveHistory.Keys.SESSION_ID.name(), ss.getSessionId());
```

Comment:
use a better name, like `queryStats`.

variable naming in Hive source code is very casual (which I consider bad). Don't follow the coding style.